notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,471 @@
+
+
+
+## Example: Model Selection for Ice Cream Ratings
+We will use the complete model selection process, including cross-validation, to select a model that predicts ice cream ratings from ice cream sweetness. The complete ice cream dataset and a scatter plot of the overall rating versus ice cream sweetness are shown below.
+
+
+```python
+# HIDDEN
+ice = pd.read_csv('icecream.csv')
+transformer = PolynomialFeatures(degree=2)
+X = transformer.fit_transform(ice[['sweetness']])
+
+clf = LinearRegression(fit_intercept=False).fit(X, ice[['overall']])
+xs = np.linspace(3.5, 12.5, 300).reshape(-1, 1)
+rating_pred = clf.predict(transformer.transform(xs))
+
+#ice.plot.scatter('sweetness', 'overall')
+#plt.plot(xs, rating_pred)
+#plt.title('Degree 2 polynomial fit');
+
+temp = pd.DataFrame(xs, columns = ['sweetness'])
+temp['overall'] = rating_pred
+
+np.random.seed(42)
+x_devs = np.random.normal(scale=0.2, size=len(temp))
+y_devs = np.random.normal(scale=0.2, size=len(temp))
+temp['sweetness'] = np.round(temp['sweetness'] + x_devs, decimals=2)
+temp['overall'] = np.round(temp['overall'] + y_devs, decimals=2)
+
+ice = pd.concat([temp, ice])
+```
+
+
+```python
+ice
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>sweetness</th>
+      <th>overall</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>3.60</td>
+      <td>3.09</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>3.50</td>
+      <td>3.17</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>3.69</td>
+      <td>3.46</td>
+    </tr>
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>11.00</td>
+      <td>5.90</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>11.70</td>
+      <td>5.50</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>11.90</td>
+      <td>5.40</td>
+    </tr>
+  </tbody>
+</table>
+<p>309 rows × 2 columns</p>
+</div>
+
+
+
+
+```python
+# HIDDEN
+plt.scatter(ice['sweetness'], ice['overall'], s=10)
+plt.title('Ice Cream Rating vs. Sweetness')
+plt.xlabel('Sweetness')
+plt.ylabel('Rating');
+```
+
+
+![png](bias_cv_files/bias_cv_9_0.png)
+
+
+Using degree 10 polynomial features on the dataset below results in a perfectly accurate model for the 10 datapoints. Unfortunately, this model fails to generalize to previously-unseen data from the population.
```

Comment:
Which 10 data points are you talking about? I am a bit confused since there are 309 rows in the "ice" table but only 9 data points in the csv file.

Space between data and points in "datapoints"
Take out dash between "previously" and "unseen"